doc/README-mosquitto.md

```diff
@@ -0,0 +1,151 @@
+its only required input. 
+
+## Usage
+
+```
+❯ mosquitto2john.py --help
+usage: mosquitto2john.py [-h] [-hc] [-o OUT_FILE] passwd_file
+
+positional arguments:
+  passwd_file           Path to the target mosquitto_passwd file.
+
+optional arguments:
+  -h, --help            show this help message and exit
+  -hc, --hashcat        Convert hashes to hashcat friendly formats.
+  -o OUT_FILE, --out-file OUT_FILE
+                        File name for converted hashes - required only for 
+			mixed hashes.
+
+```
+
+Whilst options are available for hashcat conversion and out_file writing, the
+script is intended to work in the classic something2john.py way, piping Stdout 
+straight into an outfile of your choosing:
+
+```
+❯ mosquitto2john.py mosquitto_passwd > mosquitto.hash
+❯ john ./mosquitto.hash
+[...SNIP...]
+Session completed. 
+```
+### Hashcat
+
+The hashcat option converts the hashes to Hashcat friendly formats - mode 1710 
+for SHA512 and 12100 for pbkdf2-hmac-sha512 hashes. In the following example, 
+some SHA512 and pbkdf2-hmac-sha512 hashes have already been exported into 
+separate files according to their type:
+
+```
+❯ ls
+sha512_hashes     HMAC_hashes
+# Identify the SHA512 hashes by their format: HEX_DIGEST:HEX_SALT
+❯ head -n 1 sha512_hashes
+SOMELONGHEXDIGEST:SOMEHEXSALT
+❯ hashcat -a 0 -m 1710 --hex-salt ./sha512_hashes [WORDLIST]
+[... SNIP ...]
+# Identify the HMAC hashes by a leading sha512: and following iteration count
+# These also have base64 salt and digests
+❯ head -n 1 HMAC_hashes
+sha512:101:BaSe64SaLt:BaSe64DiGeSt==
+❯ hashcat -a 0 -m 12100 ./HMAC_hashes [WORDLIST]
+[... SNIP ...]
+```
+
+**Note** the *use of --hex-salt in mode 1710* this is required for successful
+cracking.
+
+## Troubleshooting
+
+### No hashes loaded?
+
+If your individual install isn't automatically detecting the hash type, you 
+can try manually checking the hash format and specifying it with *--format*
+on the John command line:
+
+```
+# The format will be just after the username field
+❯ head -n 1 ./mosquitto.hash
+username:$dynamic_82$SOMELONGHASH$SOMESALT	
+# Now manually specify the dynamic_82 format
+❯ john ./mosquitto.hash --format=dynamic_82
+[...SNIP...]
+Session completed. 
+
+# Again for the pbkdf2-hmac-sha512 format
+❯ head -n 1 ./different_mosquitto.hash
+username2:$pbkdf2-hmac-sha512$101$.SOMESALT.SOMELONGHASH	
+# Now manually specify the pbkdf2-hmac-sha512 format
+❯ john ./mosquitto.hash --format=pbkdf2-hmac-sha512
+[...SNIP...]
+Session completed. 
+```
+
+### Mixed Hash Types?
+
+The Eclipse mosquitto_passwd program might allow different hash varieties, 
+depending on the Version you're working against. In the rare case that this
+could come up, you'll see an error and will be prompted to specify an 
+`out_file_name`; the script will then split the hash types out for you to 
+crack independently from one another. Note that the file extensions 
+allude to the hash format, handy if you want to use the --format flag in
+John. 
```

Comment:
Actually, John is happy with a shared file, where the `--format` option would be used to choose which of those to load. If it were not for hashcat, I'd have suggested dropping the out-file support.

Reply:
Interesting point, I just played around in hashcat and it's happy-ish with a shared file - it throws a token length exception but then continues to work on the 'good' hash. I'm tempted based on what you said, and for simplicitys sake, to remove this functionality outright; I guess in that instance, the only thing helping the user identify this mixed hash case is the README, so I could add some epilogue to the usage message pointing to the JtR docs. Does that sound reasonable?

Reply:
I'm fine dropping the out-file support altogether. My concern about hashcat was that it could perhaps flood with that token length exception when there are many hashes of the other type. John prints a warning suggesting use of the `--format` option just once per another detected format. Anyway, I think this flood issue is very minor - there probably won't be that many accounts in `mosquitto_passwd` files, and it's not hard to use grep to choose just one hash type if needed. So yes, please go ahead and simplify this. Thanks!

Reply:
Great, I've got this sorted now. 

The only other thing I thought of is printing a warning via stderr to the effect "careful you have mixed hash formats", but I've left it out for now, in favour of a pointer in the usage message to the README for now. Both John and Hashcat will give plenty of visual output and I do think this will be a rare issue, so hopefully that will be adequate.